application/src/main/java/org/thingsboard/server/service/entitiy/alarm/DefaultTbAlarmService.java

```diff
@@ -93,6 +94,34 @@ public ListenableFuture<Void> clear(Alarm alarm, User user) {
         }, MoreExecutors.directExecutor());
     }
 
+    @Override
+    public ListenableFuture<Void> assign(Alarm alarm, User user, UserId assigneeId) {
+        long assignTs = System.currentTimeMillis();
+        ListenableFuture<Boolean> future = alarmSubscriptionService.assignAlarm(alarm.getTenantId(), alarm.getId(), assigneeId, assignTs);
```

Comment:
This should return saved alarm. Then we should use the saved alarm for notification. We can't duplicate: 
 
```
                alarm.setAssignTs(assignTs);
                alarm.setAssigneeId(assigneeId);
```